### PR TITLE
Migrate GProjectDialog to script setup

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
Migrate GProjectDialog to script setup. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Use `Hide whitespace` for review.

**Release note**:

```bugfix user

```

### CODE REVIEW

**Review by grolu — APPROVED** [review]
/lgtm

File: frontend/src/components/dialogs/GProjectDialog.vue [hunk 1]
```diff
@@ -10,90 +10,83 @@ SPDX-License-Identifier: Apache-2.0
...
+
+        <v-row v-if="costObjectSettingEnabled">
+          <v-col cols="12">
+            <v-text-field
+              v-model="v$.costObject.$model"
+              variant="underlined"
+              color="primary"
+              :label="costObjectTitle"
+              :error-messages="getErrorMessages(v$.costObject)"
+              @update:model-value="v$.costObject.$touch()"
```
[the remark this answers is not in the document]
Comment (petersutter) [reply frontend/src/components/dialogs/GProjectDialog.vue:45]: ✅

File: frontend/src/components/dialogs/GProjectDialog.vue [hunk 2]
```diff
@@ -10,90 +10,83 @@ SPDX-License-Identifier: Apache-2.0
...
+          <v-col cols="12">
+            <v-text-field
+              ref="refProjectName"
+              v-model.trim="v$.projectName.$model"
+              variant="underlined"
+              color="primary"
+              label="Name"
+              counter="10"
+              :error-messages="getErrorMessages(v$.projectName)"
+              @update:model-value="v$.projectName.$touch()"
```
[the remark this answers is not in the document]
Comment (petersutter) [reply frontend/src/components/dialogs/GProjectDialog.vue:31]: ✅

File: frontend/src/components/dialogs/GProjectDialog.vue [hunk 3]
```diff
@@ -10,90 +10,83 @@ SPDX-License-Identifier: Apache-2.0
...
+              @blur="v$.projectName.$touch()"
```
[the remark this answers is not in the document]
Comment (petersutter) [reply frontend/src/components/dialogs/GProjectDialog.vue:32]: no, will remove it ✅

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**